forum/qa-plugin/user-activity-log/user-activity-log-search.php

```diff
@@ -0,0 +1,304 @@
+<?php
+
+class user_activity_search
+{
+    
+    private $searchArr;
+    private $dbResult;
+
+    public function match_request(string $request) 
+    {
+        $_SESSION['query'] = qa_sanitize_html($_POST['request']) ?? null;
+        $_SESSION['condition'] = qa_sanitize_html($_POST['condition']) ?? null;
+        $_SESSION['date'] = qa_sanitize_html($_POST['date'])?? null;
+        $_SESSION['resultsCount'] = qa_sanitize_html($_POST['resultsCount']) ?? null;
+        $_SESSION['sortBy'] = qa_sanitize_html($_POST['fromOldest']) ?? null;
+
+        $this->searchArr = [
+            'request' => $_SESSION['query'],
+            'condition' => $_SESSION['condition'],
+            'date' => $_SESSION['date'], 
+            'resultsCount' => $_SESSION['resultsCount'], 
+            'fromOldest' => $_SESSION['sortBy']
```

Comment:
tak sobie myślę, że w przypadku sesyjnych zmiennych warto sanityzować je przy zapisie jak i przy odczycie. Jasne, to trochę może być tu nad wyraz, ale tak ja staram się u nas w robocie postępowac.